addon/components/number-input.js

```diff
@@ -32,7 +32,11 @@ export default AbstractInput.extend({
    * @returns {Number} parse value
    */
   parseValue (value) {
-    const number = parseFloat(value)
-    return _.isFinite(number) ? number : null
+    let result = null
+    if (value !== undefined && value !== null) {
```

Comment:
Not sure you added this additional check as `parseFloat(null)` and `parseFloat(undefined)` both return `NaN`. You should be able to just do:
```javascript
    const number = parseFloat(value)
   return _.isFinite(number) ? number : null
```